packages/@react-aria/overlays/src/calculatePosition.ts

```diff
@@ -338,9 +338,29 @@ export function calculatePositionInternal(
   delta = getDelta(crossAxis, position[crossAxis], overlaySize[crossSize], boundaryDimensions, padding);
   position[crossAxis] += delta;
 
```

Comment:
Was tinkering around and found a possible alternative approach (not sure if better or not, its pretty rough):
The approach is to check where the popover is going to placed relative to the button and based on that whether the trigger button center is too close to the top/bottom/left/right of the boundary element viewport. If it is too close, remove the container padding from the overlay position calculations.

```
 // NOTE this code is called before the "delta" is calculated on line 323
  let boundaryLimits = {
    top: boundaryDimensions.top + boundaryDimensions.scroll.top,
    bottom: boundaryDimensions.top + boundaryDimensions.scroll.top + boundaryDimensions.height,
    left: boundaryDimensions.left + boundaryDimensions.scroll.left,
    right: boundaryDimensions.left + boundaryDimensions.scroll.left + boundaryDimensions.width
  };

  if (placementInfo.placement === 'left' || placementInfo.placement === 'right') {
    let centerOfButtonVertical = childOffset.top + childOffset.height / 2;
    // Really it should be center of button center + padding + half of arrow width > boundary but we don't have access to arrow dimensions
    if (centerOfButtonVertical + padding * 2 > boundaryLimits.bottom || centerOfButtonVertical - padding * 2 < boundaryLimits.top) {
      padding = 0;
    }
  } else if (placementInfo.placement === 'top' || placementInfo.placement === 'bottom') {
    let centerOfButtonHorizontal = childOffset.left + childOffset.width / 2;
    if (centerOfButtonHorizontal + padding * 2 > boundaryLimits.right || centerOfButtonHorizontal - padding * 2 < boundaryLimits.left) {
      padding = 0;
    }
  }
```
Problems with my approach: 
- we don't have the arrow tip dimensions
- could be improved by attempting to keep the arrow tip in view via modifying the arrowPosition